qubit-sdk/src/main/java/com/qubit/android/sdk/internal/callbacktracker/repository/CallbackRequestRepositoryImpl.kt

```diff
@@ -0,0 +1,60 @@
+package com.qubit.android.sdk.internal.callbacktracker.repository
+
+import android.annotation.SuppressLint
+import android.content.Context
+
+@SuppressLint("ApplySharedPref")
+internal class CallbackRequestRepositoryImpl(
+    appContext: Context
+) : CallbackRequestRepository {
+
+  companion object {
+    private const val PREFERENCES_FILE = "qubit_callback_requests"
+    private const val DEFAULT_KEY_VALUE = 1L
+  }
+
+  private val sharedPreferences = appContext.getSharedPreferences(PREFERENCES_FILE, Context.MODE_PRIVATE)
+  private var currentKey = DEFAULT_KEY_VALUE
+
+  override fun init() {
+    val lastKey = getLastKey()
+    currentKey = lastKey ?: DEFAULT_KEY_VALUE
+  }
+
+  @Synchronized
+  override fun insert(url: String) {
+    currentKey++
+    sharedPreferences.edit()
+        .putString(currentKey.toString(), url)
+        .commit()
+  }
+
+  @Synchronized
+  override fun fetchFirst(): String? {
```

Comment:
It can be written this way:
```
  @Synchronized
  override fun fetchFirst(): String? =
    getFirstKey()?.toString()?.let { key -> 
      sharedPreferences.getString(key, null)
        ?.also { remove(key) } 
    }
```